### PR TITLE
Make the Finalize method of the creator public

### PR DESCRIPTION
**Pull request by adrg**
This change is [Reviewable](https://reviewable.io/reviews/unidoc/unipdf/198)

### CODE REVIEW

File: creator/creator.go [hunk 1]
```diff
@@ -346,9 +346,16 @@ func (c *Creator) Context() DrawContext {
...
+// of adding headers and footers, as well as generating the front page,
+// table of contents and outlines.
+// Finalize is automatically called before writing the document out. Calling the
+// method manually can be useful when adding external pages to the creator,
+// using the AddPage method, as it renders all creator blocks to the added
+// pages, without having to write the document out.
+// NOTE: TOC and outlines are generated only if the AddTOC and AddOutlines
+// fields of the creator are set to true (enabled by default).
+func (c *Creator) Finalize() error {
 	totPages := len(c.pages)
```
Comment (gunnsth) [inline creator/creator.go]: Should ignore if already finalized (c.finalized is true) ?

File: creator/creator.go [hunk 2]
```diff
@@ -346,9 +346,16 @@ func (c *Creator) Context() DrawContext {
...
-// as generating front Page and table of contents.
-func (c *Creator) finalize() error {
+// Finalize renders all blocks to the creator pages. In addition, it takes care
+// of adding headers and footers, as well as generating the front page,
+// table of contents and outlines.
+// Finalize is automatically called before writing the document out. Calling the
+// method manually can be useful when adding external pages to the creator,
+// using the AddPage method, as it renders all creator blocks to the added
+// pages, without having to write the document out.
+// NOTE: TOC and outlines are generated only if the AddTOC and AddOutlines
```
Comment (gunnsth) [inline creator/creator.go]: If there is no content in the TOC/outlines, it does not get added either, might be worth mentioning that?

File: creator/creator.go (hunk 1 above)
Comment (adrg) [reply creator/creator.go]: Yes, that should be the case. Good catch, thanks.

File: creator/creator.go (hunk 2 above)
Comment (adrg) [reply creator/creator.go]: Sure. I adapted the note to also include the ways to add content to TOCs/outlines.

**Pull request merged**